package/MDAnalysis/core/universe.py

```diff
@@ -1426,16 +1464,69 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
         if generate_coordinates:
             if not addHs:
                 raise ValueError("Generating coordinates requires adding "
-                "hydrogens with `addHs=True`")
+                                 "hydrogens with `addHs=True`")
 
             numConfs = rdkit_kwargs.pop("numConfs", numConfs)
-            if not (type(numConfs) is int and numConfs > 0):
+            if not (isinstance(numConfs, int) and numConfs > 0):
                 raise SyntaxError("numConfs must be a non-zero positive "
-                "integer instead of {0}".format(numConfs))
+                                  "integer instead of {0}".format(numConfs))
             AllChem.EmbedMultipleConfs(mol, numConfs, **rdkit_kwargs)
 
         return cls(mol, **kwargs)
 
+    def guess_TopologyAttributes(self, context=None, to_guess=None, **kwargs):
+        """guess attributes passed to the universe within specific context
+
+        Parameters
+        ----------
+        context: string or Guesser class
+        for calling a matching guesser class for this specific context
+        to_guess: list
+        list of atrributes to be guessed then added to the universe
+        **kwargs: extra arguments are passed to the guesser class
+        
+        Examples
+        --------
+        guess masses and elements attribute 
+        u.guess_TopologyAttributes(context='default', to_guess=['masses', elements])
+        """
+        if not context:
+            context =  self._context
+                
+        guesser = get_guesser(context, self.universe, **kwargs)
+        self._context = guesser
+
+        if guesser.is_guessable(to_guess):           
+            if 'bonds' in to_guess:
+                self._kwargs['guess_bonds'] = True
+                to_guess.remove('bonds')
+
+            for attr in to_guess:
+                if any(attr == a for a in self._topology_atrrs):
+                    warnings.warn('The attribute {} have already been read '
+                                  'from the topology file. You are trying to '
+                                  'overwrite it by guessed values'
+                                  .format(attr))
+
+                values = guesser.guess_Attr(attr)
+                tcls = _TOPOLOGY_ATTRS[attr](values, True)
+                self.add_TopologyAttr(tcls)
+                logger.info('attribute {} has been guessed successfully.'.format(attr))
+
+            if self._kwargs['guess_bonds']:
+                if 'bonds' in self._topology_atrrs:
+                    warnings.warn('The attribute bonds have already been read '
+                                  'from the topology file. You are trying to '
+                                  'overwrite it by guessed values'
+                                  )
+                
+                self.atoms.guess_bonds(self._kwargs['vdwradii'], guesser)
```

Comment:
tbh I didn't get why bond guessing is attached to the `AtomGroup`, so I added context checking inside the `guess_bonds` from the `AtomGroup` to not change the current bond guessing methodology.  but I think it would be better to use a `guess_bonds `method directly from a guesser. If it is ok, I could work on this

Reply:
tbh I didn't get why bond guessing is attached to the `AtomGroup`, so I added context checking inside the `guess_bonds` from the `AtomGroup` to not change the current bond guessing methodology.  but I think it would be better to use a `guess_bonds `method directly from a guesser. If it is ok, I could work on this